### PR TITLE
feat: add auto run v8 ci

### PR DESCRIPTION
**Pull request by gengjiawen**
fix: https://github.com/nodejs/node-core-utils/issues/559
test pr: https://github.com/nodejs/node/pull/44986

Hopefully this will make us test v8 easier :)

local test
```
node bin/ncu-ci.js run 44986
```

currently I run into Jenkins credentials invalid. But I have used this credential before, and it worked.

cc @targos @richardlau @nodejs/v8-update

### CODE REVIEW

File: lib/ci/run_ci.js [hunk 1]
```diff
@@ -71,6 +76,28 @@ export class RunPRJob {
...
+      await this.prData.getPR();
+      const labels = this.prData.pr.labels
+      if (labels.includes('v8 engine')) {
+        cli.startSpinner('Starting V8 CI job');
+        const response = await this.request.fetch(CI_V8_URL, {
+          method: 'POST',
+          headers: {
+            'Jenkins-Crumb': crumb
+          },
+          body: this.payload
```
Comment (richardlau) [inline lib/ci/run_ci.js:90]: This won't work for the V8 CI job as it takes different parameters to the pull request job.

**Review by richardlau — CHANGES_REQUESTED** [review]
Please undo the style changes causing the linter to fail. Also this doesn't work:
```
$ git rev-parse --short HEAD
8e866ef
$ node bin/ncu-ci.js --owner nodejs --repo node run 44961
✔  Jenkins credentials valid
⠋ Starting PR CI job(node:1644671) [https://github.com/node-fetch/node-fetch/issues/1167] DeprecationWarning: form-data doesn't follow the spec and requires special treatment. Use alternative package
(Use `node --trace-deprecation ...` to show where the warning was created)
✔  PR CI job successfully started
✖  Failed to start CI
$

```

**Comment by richardlau** [comment]
> ```
> $ git rev-parse --short HEAD
> 8e866ef
> $ node bin/ncu-ci.js --owner nodejs --repo node run 44961
> ✔  Jenkins credentials valid
> ⠋ Starting PR CI job(node:1644671) [https://github.com/node-fetch/node-fetch/issues/1167] DeprecationWarning: form-data doesn't follow the spec and requires special treatment. Use alternative package
> (Use `node --trace-deprecation ...` to show where the warning was created)
> ✔  PR CI job successfully started
> ✖  Failed to start CI
> $
> ```

More info on the failure -- the "Failed to start CI" message is from the try-catch handler https://github.com/nodejs/node-core-utils/blob/8e866ef03f9ae21301024eaf22db87eb212a58b2/lib/ci/run_ci.js#L102
With additional debug:
```diff
diff --git a/lib/ci/run_ci.js b/lib/ci/run_ci.js
index e850a7a..51d6fb4 100644
--- a/lib/ci/run_ci.js
+++ b/lib/ci/run_ci.js
@@ -99,6 +99,7 @@ export class RunPRJob {
       }

     } catch (err) {
+      console.error(err);
       cli.stopSpinner('Failed to start CI', this.cli.SPINNER_STATUS.FAILED);
       return false;
     }
```
the error is:
```
$ node bin/ncu-ci.js --owner nodejs --repo node run 44986
✔  Jenkins credentials valid
✔  PR CI job successfully started
TypeError: Cannot read property 'gql' of undefined
    at PRData.getPR (file:///home/rlau/sandbox/github/node-core-utils/lib/pr_data.js:80:34)
    at RunPRJob.start (file:///home/rlau/sandbox/github/node-core-utils/lib/ci/run_ci.js:81:25)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
✖  Failed to start CI
```

**Comment by gengjiawen** [comment]
Is the credential put in `~/.ncurc`

**Comment by richardlau** [comment]
> Is the credential put in `~/.ncurc`

Yes. Credential is working -- the pull request job was successfully started and it's the additions in this PR that are not working.

**Comment by gengjiawen** [comment]
> > Is the credential put in `~/.ncurc`
> 
> Yes. Credential is working -- the pull request job was successfully started and it's the additions in this PR that are not working.

Yeap. On my local I stuck at credential issue. So I mark it as draft.

**Comment by gengjiawen** [comment]
now jenkins with 500. Can you check the log @richardlau , thx.

**Comment by gengjiawen** [comment]
Now the issues is it's not auto-commenting pr ? which part should I check ?

File: lib/ci/run_ci.js [hunk 2]
```diff
@@ -71,7 +89,30 @@ export class RunPRJob {
         return false;
       }
       cli.stopSpinner('PR CI job successfully started');
+
+      // check if the job need a v8 build and trigger it
+      await this.prData.getPR();
+      const labels = this.prData.pr.labels;
+      console.log(labels);
```
Comment (richardlau) [inline lib/ci/run_ci.js:96]:
```suggestion
```

Or if you really want to keep it, move to `debuglog`.

**Comment by gengjiawen** [comment]
please squash this commit for conventional commit.

**Comment by legendecas** [comment]
> @gengjiawen: 
> @legendecas Is this because proxy issue, not supporting socks5 proxy ?
> 
> Update: a China only issue, fucking gfw. `proxy-agent` looks only work in http proxy.

Not sure what I can help here?

**Comment by gengjiawen** [comment]
> > @gengjiawen:
> > @legendecas Is this because proxy issue, not supporting socks5 proxy ?
> > Update: a China only issue, fucking gfw. `proxy-agent` looks only work in http proxy.
> 
> Not sure what I can help here?

fixed, proxy compatibility issue

**Comment by gengjiawen** [comment]
@joyeecheung Can you review this ?

**Comment by gengjiawen** [comment]
@aduh95 Can you take a look at this ? thx

File: .gitignore [hunk 3]
```diff
@@ -8,3 +8,4 @@ tmp-*
 .eslintcache
 .ncu
 package-lock.json
+.idea/
```
Comment (Trott) [inline .gitignore:11]: I'm not blocking on this but this seems unrelated and perhaps should go in the user's own `.gitignore`?

**Comment by Trott** [comment]
This looks good to me. Is there anything we can/should do about the small decrease in test coverage?  I'm not worried about it if there isn't much to do.

File: .npmrc [hunk 4]
```diff
@@ -1 +1,2 @@
 package-lock=false
+auto-install-peers=true
```
Comment (Trott) [inline .npmrc:2]: Is this change needed?

**Comment by Trott** [comment]
> please squash this commit for conventional commit.

Do you want to squash and force push to your branch?

File: .npmrc (hunk 4 above)
Comment (gengjiawen) [reply .npmrc:2]: Yeap. Current config not working on Pnpm.

**Comment by gengjiawen** [comment]
> This looks good to me. Is there anything we can/should do about the small decrease in test coverage? I'm not worried about it if there isn't much to do.

It's a duplicate logic already tested (different URL and HTTP payload for v8 jenkins task). I think in this case, it's okay.

**Comment by gengjiawen** [comment]
@Trott Can you merge this and https://github.com/nodejs/github-bot/pull/353 (already reviewed by @joyeecheung ) ? thx

File: .gitignore [hunk 5]
```diff
@@ -8,3 +8,4 @@ tmp-*
 .eslintcache
 .ncu
 package-lock.json
+
```
Comment (Trott) [inline .gitignore:11]:
```suggestion
```
